### PR TITLE
Rooms that contain two participants

### PR DESCRIPTION
**Pull request by NicoleRauch**

### CODE REVIEW

File: socrates/lib/activities/rooms.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
+'use strict';
+
+var _ = require('lodash');
+var WaitinglistEntry = require('simple-configure').get('beans').get('waitinglistEntry');
+
+function Rooms(roomsObject, allKnownMemberIds) {
+  console.log(roomsObject);
```
Comment (leider) [inline socrates/lib/activities/rooms.js]: ooopsi. console-log soll doch wohl raus...

Comment (NicoleRauch) [inline socrates/lib/activities/rooms.js]: Hehe. Well spotted :-)

File: socrates/lib/activities/socratesActivitiesService.js [hunk 2]
```diff
@@ -119,6 +119,28 @@ module.exports = {
...
+
+  newParticipantPairFor: function (resourceName, participant1Nick, participant2Nick, callback) {
+    var self = this;
+    activitystore.getActivity(currentUrl, function (err, activity) {
+      if (err || !activity) { return callback(err); }
+      memberstore.getMember(participant1Nick, function (err, participant1) {
+        if (err || !participant1) { return callback(err); }
+        memberstore.getMember(participant2Nick, function (err, participant2) {
+          if (err || !participant2) { return callback(err); }
+          activity.socratesResourceNamed(resourceName).rooms().add(participant1.id(), participant2.id());
```
Comment (leider) [inline socrates/lib/activities/socratesActivitiesService.js]: If you create a method activity.rooms(resourceName), use it!
(see below)

File: socrates/lib/registration/index.js [hunk 3]
```diff
@@ -210,15 +210,54 @@ app.get('/management', function (req, res, next) {
...
-            res.render('managementTables', {
-              activity: activity,
-              addonLines: addonLines,
-              waitinglistLines: waitinglistLines,
-              addonLinesOfUnsubscribedMembers: [],
-              tshirtsizes: managementService.tshirtSizes(addonLines),
-              durations: managementService.durations(activity),
-              formatDates: formatDates,
-              formatList: formatList
+            memberstore.getMembersForIds(activity.socratesResourceNamed('bed_in_double').rooms().participantsWithoutRoom(), function (err, unpairedDoubleParticipants) {
```
Comment (leider) [inline socrates/lib/registration/index.js]: I#d like to see async used here

File: socrates/lib/activities/rooms.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+Rooms.prototype.participantsInRoom = function () {
+  var participantsInRoom = [];
+  participantsInRoom.push(_.pluck(this.state, 'participant1'));
+  participantsInRoom.push(_.pluck(this.state, 'participant2'));
+  return _.flatten(participantsInRoom);
+};
+
+Rooms.prototype.roomPairs = function () {
+  return this.state.slice(); // attention - this is a shallow copy only!
```
Comment (leider) [inline socrates/lib/activities/rooms.js]: doesn't splice change the array it is called on?

Comment (NicoleRauch) [inline socrates/lib/activities/rooms.js]: That's why I'm invoking slice() here (without the "p") ...

**Pull request merged**